### DISCUSSION TITLE
RTMP to RTSP ERR: wrong interleaved ids, expected

### DISCUSSION THREAD

**Discussion by t3therdev** [bluenviron/mediamtx#295] [Questions]
## Which version are you using?

v0.15.1

## Which operating system are you using?

OS

- [ ] Linux
- [X] Windows
- [ ] macOS

## Describe the issue

Pushing RTMP from DJI Go 4 App (Android) to RTSPSS. RTMP to RTSP. VLC and FFplay work, but other ffmpeg based clients we have do not. 

Standard config, no special settings. 

Errors are showing like this on RTSPS log:

2021/03/25 18:16:40 [I] [1/0/0] [client XX.XX.XX.XX:18246] connected (RTMP)
2021/03/25 18:16:41 [I] [1/1/0] [client XX.XX.XX.XX:18246] is publishing to path 'drone', 2 tracks
2021/03/25 18:16:50 [I] [2/1/0] [client XX.XX.XX.XX:12296] connected (RTSP/TCP)
2021/03/25 18:16:50 [I] [2/1/0] [client XX.XX.XX.XX:12296] ERR: wrong interleaved IDs, expected [2 3], got [0 1]
2021/03/25 18:16:50 [I] [1/1/0] [client XX.XX.XX.XX:12296] disconnected

**Comment by aler9** [comment]
Hello, please post the log with the parameter `logLevel: debug` in the configuration, thanks

**Comment by t3therdev** [comment]
`2021/03/26 08:12:49 [I] [0/0/0] rtsp-simple-server v0.15.1
2021/03/26 08:12:49 [I] [0/0/0] [RTSP/TCP listener] opened on :8554
2021/03/26 08:12:49 [I] [0/0/0] [RTSP/TLS listener] opened on :8555
2021/03/26 08:12:49 [I] [0/0/0] [RTMP listener] opened on :1935
2021/03/26 08:15:45 [I] [1/0/0] [client XX.XX.XX.XX:23931] connected (RTMP)
2021/03/26 08:15:45 [I] [1/1/0] [client XX.XX.XX.XX:23931] is publishing to path 'drone', 2 tracks
2021/03/26 08:16:03 [I] [2/1/0] [client XX.XX.XX.XX:51424] connected (RTSP/TCP)
2021/03/26 08:16:03 [D] [2/1/0] [client XX.XX.XX.XX:51424] [c->s] OPTIONS rtsp://XX.XX.XX.XX:8554/drone RTSP/1.0
CSeq: 1
User-Agent: RtspClientSharp


2021/03/26 08:16:03 [D] [2/1/0] [client XX.XX.XX.XX:51424] [s->c] RTSP/1.0 200 OK
CSeq: 1
Public: DESCRIBE, ANNOUNCE, SETUP, PLAY, RECORD, PAUSE, GET_PARAMETER, TEARDOWN
Server: gortsplib


2021/03/26 08:16:03 [D] [2/1/0] [client XX.XX.XX.XX:51424] [c->s] DESCRIBE rtsp://xx.xx.xx.xx:8554/drone RTSP/1.0
Accept: application/sdp
CSeq: 2
User-Agent: RtspClientSharp


2021/03/26 08:16:03 [D] [2/1/0] [client XX.XX.XX.XX:51424] [s->c] RTSP/1.0 200 OK
CSeq: 2
Content-Base: rtsp://XX.XX.XX.XX:8554/drone/
Content-Length: 437
Content-Type: application/sdp
Server: gortsplib

v=0
o=- 0 0 IN IP4 127.0.0.1
s=Stream
c=IN IP4 0.0.0.0
t=0 0
m=video 0 RTP/AVP 96
a=rtpmap:96 H264/90000
a=fmtp:96 packetization-mode=1; sprop-parameter-sets=Z0KAH9oBQBbpSCgwMDaFCag=,aM4G4g==; profile-level-id=42801F
a=control:trackID=0
m=audio 0 RTP/AVP 96
a=rtpmap:96 MPEG4-GENERIC/44100/1
a=fmtp:96 profile-level-id=1; mode=AAC-hbr; sizelength=13; indexlength=3; indexdeltalength=3; config=120856e500
a=control:trackID=1

2021/03/26 08:16:03 [D] [2/1/0] [client XX.XX.XX.XX:51424] [c->s] SETUP rtsp://XX.XX.XX.XX:8554/drone/trackID=1 RTSP/1.0
CSeq: 3
Transport: RTP/AVP/TCP;unicast;interleaved=0-1
User-Agent: RtspClientSharp


2021/03/26 08:16:03 [D] [2/1/0] [client XX.XX.XX.XX:51424] [s->c] RTSP/1.0 400 Bad Request
CSeq: 3
Server: gortsplib


2021/03/26 08:16:03 [I] [2/1/0] [client XX.XX.XX.XX:51424] ERR: wrong interleaved IDs, expected [2 3], got [0 1]
2021/03/26 08:16:03 [I] [1/1/0] [client XX.XX.XX.XX:51424] disconnected
2021/03/26 08:16:09 [I] [2/1/0] [client XX.XX.XX.XX:31505] connected (RTSP/TCP)
2021/03/26 08:16:09 [D] [2/1/0] [client XX.XX.XX.XX:31505] [c->s] OPTIONS rtsp://XX.XX.XX.XX:8554/drone RTSP/1.0
CSeq: 1
User-Agent: RtspClientSharp


2021/03/26 08:16:09 [D] [2/1/0] [client XX.XX.XX.XX:31505] [s->c] RTSP/1.0 200 OK
CSeq: 1
Public: DESCRIBE, ANNOUNCE, SETUP, PLAY, RECORD, PAUSE, GET_PARAMETER, TEARDOWN
Server: gortsplib


2021/03/26 08:16:09 [D] [2/1/0] [client XX.XX.XX.XX:31505] [c->s] DESCRIBE rtsp://XX.XX.XX.XX:8554/drone RTSP/1.0
Accept: application/sdp
CSeq: 2
User-Agent: RtspClientSharp


2021/03/26 08:16:09 [D] [2/1/0] [client XX.XX.XX.XX:31505] [s->c] RTSP/1.0 200 OK
CSeq: 2
Content-Base: rtsp://XX.XX.XX.XX:8554/drone/
Content-Length: 437
Content-Type: application/sdp
Server: gortsplib

v=0
o=- 0 0 IN IP4 127.0.0.1
s=Stream
c=IN IP4 0.0.0.0
t=0 0
m=video 0 RTP/AVP 96
a=rtpmap:96 H264/90000
a=fmtp:96 packetization-mode=1; sprop-parameter-sets=Z0KAH9oBQBbpSCgwMDaFCag=,aM4G4g==; profile-level-id=42801F
a=control:trackID=0
m=audio 0 RTP/AVP 96
a=rtpmap:96 MPEG4-GENERIC/44100/1
a=fmtp:96 profile-level-id=1; mode=AAC-hbr; sizelength=13; indexlength=3; indexdeltalength=3; config=120856e500
a=control:trackID=1

2021/03/26 08:16:09 [D] [2/1/0] [client XX.XX.XX.XX:31505] [c->s] SETUP rtsp://XX.XX.XX.XX:8554/drone/trackID=1 RTSP/1.0
CSeq: 3
Transport: RTP/AVP/TCP;unicast;interleaved=0-1
User-Agent: RtspClientSharp


2021/03/26 08:16:09 [D] [2/1/0] [client XX.XX.XX.XX:31505] [s->c] RTSP/1.0 400 Bad Request
CSeq: 3
Server: gortsplib


2021/03/26 08:16:09 [I] [2/1/0] [client XX.XX.XX.XX:31505] ERR: wrong interleaved IDs, expected [2 3], got [0 1]
2021/03/26 08:16:09 [I] [1/1/0] [client XX.XX.XX.XX:31505] disconnected
2021/03/26 08:16:14 [I] [2/1/0] [client XX.XX.XX.XX:27717] connected (RTSP/TCP)
2021/03/26 08:16:14 [D] [2/1/0] [client XX.XX.XX.XX:27717] [c->s] OPTIONS rtsp://XX.XX.XX.XX:8554/drone RTSP/1.0
CSeq: 1
User-Agent: RtspClientSharp


2021/03/26 08:16:14 [D] [2/1/0] [client XX.XX.XX.XX:27717] [s->c] RTSP/1.0 200 OK
CSeq: 1
Public: DESCRIBE, ANNOUNCE, SETUP, PLAY, RECORD, PAUSE, GET_PARAMETER, TEARDOWN
Server: gortsplib


2021/03/26 08:16:14 [D] [2/1/0] [client XX.XX.XX.XX:27717] [c->s] DESCRIBE rtsp://XX.XX.XX.XX:8554/drone RTSP/1.0
Accept: application/sdp
CSeq: 2
User-Agent: RtspClientSharp


2021/03/26 08:16:14 [D] [2/1/0] [client XX.XX.XX.XX:27717] [s->c] RTSP/1.0 200 OK
CSeq: 2
Content-Base: rtsp://XX.XX.XX.XX:8554/drone/
Content-Length: 437
Content-Type: application/sdp
Server: gortsplib

v=0
o=- 0 0 IN IP4 127.0.0.1
s=Stream
c=IN IP4 0.0.0.0
t=0 0
m=video 0 RTP/AVP 96
a=rtpmap:96 H264/90000
a=fmtp:96 packetization-mode=1; sprop-parameter-sets=Z0KAH9oBQBbpSCgwMDaFCag=,aM4G4g==; profile-level-id=42801F
a=control:trackID=0
m=audio 0 RTP/AVP 96
a=rtpmap:96 MPEG4-GENERIC/44100/1
a=fmtp:96 profile-level-id=1; mode=AAC-hbr; sizelength=13; indexlength=3; indexdeltalength=3; config=120856e500
a=control:trackID=1

2021/03/26 08:16:14 [D] [2/1/0] [client XX.XX.XX.XX:27717] [c->s] SETUP rtsp://XX.XX.XX.XX:8554/drone/trackID=1 RTSP/1.0
CSeq: 3
Transport: RTP/AVP/TCP;unicast;interleaved=0-1
User-Agent: RtspClientSharp


2021/03/26 08:16:14 [D] [2/1/0] [client XX.XX.XX.XX:27717] [s->c] RTSP/1.0 400 Bad Request
CSeq: 3
Server: gortsplib


2021/03/26 08:16:14 [I] [2/1/0] [client XX.XX.XX.XX:27717] ERR: wrong interleaved IDs, expected [2 3], got [0 1]
2021/03/26 08:16:14 [I] [1/1/0] [client XX.XX.XX.XX:27717] disconnected
`

**Comment by aler9** [comment]
The client (RtspClientSharp) is trying to setup the audio track before the video track - i don't know why it is doing this, but it's highly non-standard and depends on the client, not on the server.

[a comment by an automated account is not shown]
